pkg/account/api/admin_account.go

```diff
@@ -494,3 +498,154 @@ func (s *AccountService) updateLastSeen(ctx context.Context, email, organization
 
 	return s.accountStorage.UpdateAccountV2(ctx, account)
 }
+
+func (s *AccountService) GetMyOrganizationsByAccessToken(
+	ctx context.Context,
+	req *accountproto.GetMyOrganizationsByAccessTokenRequest,
+) (*accountproto.GetMyOrganizationsResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+
+	switch req.Type {
+	case accountproto.GetMyOrganizationsByAccessTokenRequest_AUTH_TYPE_UNSPECIFIED:
+		dt, err := statusInvalidOrderBy.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "Invalid auth type"),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	case accountproto.GetMyOrganizationsByAccessTokenRequest_AUTH_TYPE_GOOGLE:
+		userInfo, err := s.verifyGoogleAccessToken(ctx, req.AccessToken, localizer)
+		if err != nil {
+			return nil, err
+		}
+
+		myOrgs, err := s.getMyOrganizations(ctx, userInfo.Email, localizer)
+		if err != nil {
+			return nil, err
+		}
+
+		return &accountproto.GetMyOrganizationsResponse{Organizations: myOrgs}, nil
+	case accountproto.GetMyOrganizationsByAccessTokenRequest_AUTH_TYPE_BUCKETEER:
+		// currently we only check demo account
+		if req.Email == s.oauthConfig.DemoSignIn.Email {
+			myOrgs, err := s.getMyOrganizations(ctx, req.Email, localizer)
+			if err != nil {
+				return nil, err
+			}
+			return &accountproto.GetMyOrganizationsResponse{Organizations: myOrgs}, nil
+		}
+
+	case accountproto.GetMyOrganizationsByAccessTokenRequest_AUTH_TYPE_GITHUB:
+		// TODO: Implement GitHub access token verification
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+
+	return nil, status.Error(
+		codes.InvalidArgument,
+		localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "Invalid auth type"),
+	)
+}
+
+func (s *AccountService) verifyGoogleAccessToken(
```

Comment:
We have a similar logic in the auth/google/authenticator.go.
It will be better to reuse the authenticator to avoid future inconsistencies.
At least we should move this code to the authenticator.